ui/admin/app/controllers/scopes/scope/storage-buckets/index.js

```diff
@@ -55,6 +56,10 @@ export default class ScopesScopeStorageBucketsIndexController extends Controller
   @notifyError(({ message }) => message)
   @notifySuccess('notifications.save-success')
   async save(storageBucket) {
+    // If the role_arn is empty, then the credential type should be static
+    if (!storageBucket.role_arn) {
+      storageBucket.credentialType = TYPE_CREDENTIAL_STATIC;
+    }
```

Comment:
if user has an existing storage bucket with static credential type and edits type to dynamic credential without entering a `role_arn` value then on save it will show type change to 'dynamic' even though its still static. So this change was added so that type stays as 'static'. We already do the same for host-catalogs.

Reply:
Good catch!